src/portkey/awssig.clj

```diff
@@ -126,3 +127,75 @@
         headers (assoc headers "Authorization" auth)]
     (assoc req :headers headers)))
 
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Signing a request v2
+;; https://docs.aws.amazon.com/AmazonSimpleDB/latest/DeveloperGuide/HMACAuth.html#REST_RESTAuth
+
+(def x-amz-date-formatter-sign-2
+  (java.time.format.DateTimeFormatter/ofPattern "yyyy-MM-dd'T'HH:mm:ss'Z'"))
+
+(defn- hmac-sha-256-sign2
+   [^bytes secret ^String s]
+   (-> (javax.crypto.Mac/getInstance "HmacSHA256")
+         (doto (.init (javax.crypto.spec.SecretKeySpec. secret "HmacSHA256")))
+         (.doFinal (.getBytes s "UTF8"))))
+
+(defn- utf8-natural-byte-cmp
+   "Sort the UTF-8 query string components by parameter name with natural byte ordering.
+   I wonder if it really matters for the input we deal with"
+   [^String a ^String b]
+   (let [sa a sb b
+         a (.getBytes a "UTF8")
+         b (.getBytes b "UTF8")
+         n (min (alength a) (alength b))]
+        (loop [i 0]
+          (if (< i n)
+            (let [r (- (aget a i) (aget b i))]
+               (if (zero? r)
+                 (recur (inc i))
+                 r))
+            (- (alength a) (alength b))))))
+
+(defn- create-canonicalized-query-string-v2
+   "Step 1 : Create a canonicalized query-string for building the signature v2."
```

Comment:
I'd leave out the order of use of signature helper functions out of docstrings, order of steps should be visible from place of use already, right?

Reply:
Yes indeed but I thought repeating doc's steps was easier to understand :-)

Reply:
While I agree with @viesti, these are private vars so it doesn't bother me.

Reply:
as you prefer for this one

Reply:
Principle of least effort: don’t change anything! Kimmo, any strong argument is against?

Reply:
True. I'm ok with leaving the steps in docstrings, not necessary to do more effort now.